lib/firstline.coffee

```diff
@@ -3,38 +3,10 @@ fs = require 'fs'
 class FirstLine
     read: (filepath, callback) ->
         @callback = callback
-        fs.open filepath, 'r', (err, fd) =>
-            if (err)
-                callback err
-            else
-                @stats fd
-
-    stats: (fd) ->
-        fs.fstat fd, (err, stats) =>
-            @startReading fd, stats.size
-
-    startReading: (fd, size) ->
-        @fd = fd
-
-        @buffer = new Buffer size
-        @buffer.fill 0
-        @offset = 0
 
-        @readFile()
-
-    readFile: ->
-        length = Math.min(16, @buffer.length - @offset)
-        fs.read @fd, @buffer, @offset, length, null, @onRead
-
-    onRead: (err, bytesRead, buffer) =>
-        newline = buffer.indexOf '\n', @offset
-
-        if newline > -1
-            fs.close @fd, (err) -> console.log err if err
-            line = buffer.toString undefined, 0, newline
-            @callback null, line
-        else
-            @offset += bytesRead
-            @readFile()
+        fs.readFile filepath, (err, fileData) =>
+            if err != null
+                callback err
+            callback null, fileData.toString().match /.*/
```

Comment:
The point of this file is to read the first line of a file in the most efficient way possible. So instead of reading in the entire file, which could be very large, we instead read as few bytes as possible while looking for the the newline separator. We then pass the first line to the callback.

Your change is reading the entire file, and is actually passing a RegEx Match object, which is not what the callback is expecting, as far as I know.